### PR TITLE
add basic serializer and view

### PR DESCRIPTION
**Pull request by NikitaNikolaitsev**
Screenshot_79
Screenshot_80

### CODE REVIEW

File: book/views.py [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+from book.models import Book
+from rest_framework.decorators import action
+from book.serializers import BookSerializer
+
+
+class BookViewSet(viewsets.ModelViewSet):
+    queryset = Book.objects.all()
+    serializer_class = BookSerializer
+
+    @action(detail=True, methods=['get'], url_path='detail')
```
Comment (BornToLivee) [inline book/views.py]: Why do use @action here? Do we need additional endpoint? Also i think you need to fix flake8 and quotes

File: book/urls.py [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+from rest_framework.routers import DefaultRouter
+from django.urls import path, include
+from book.views import BookViewSet
 
-urlpatterns = []
+router = DefaultRouter()
+router.register(r'books', BookViewSet, basename='book')
+
+urlpatterns = [
+    path('', include(router.urls)),
```
Comment (BornToLivee) [inline book/urls.py]: fix quotes

File: book/views.py [hunk 3]
```diff
@@ -1,3 +1,22 @@
-from django.shortcuts import render
+from rest_framework import viewsets
+from rest_framework.response import Response
 
-# Create your views here.
+from book.models import Book
+from rest_framework.decorators import action
```
Comment (BornToLivee) [inline book/views.py]: fix imports :) Im joking(or not)

File: book/views.py [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+    serializer_class = BookSerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        book = self.get_object()
+        data = {
+            "title": book.title,
+            "author": book.author,
+            "inventory": book.inventory,
+            "daily_fee": book.daily_fee,
+            "some_additional_stat": "Example statistic"
```
Comment (BornToLivee) [inline book/views.py]: i think we dont need this "some_additional_stat"

File: book/serializers.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            raise serializers.ValidationError({"daily_fee": "Daily fee must be at least 0.09."})
+
+        if title is None:
+            raise serializers.ValidationError({"title": "Title cannot be empty."})
+
+        return attrs
+
+    class Meta:
+        model = Book
+        fields = ("title", "author", "cover", "inventory", "daily_fee")
```
Comment (BornToLivee) [inline book/serializers.py]: what about id field? do we need it?

**Review by lilarin — APPROVED** [review]
Thanks!

**Review by BornToLivee — APPROVED** [review]
GJ!

File: book/serializers.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            )
+
+        if title is None:
+            raise serializers.ValidationError(
+                {"title": "Title cannot be empty."}
+            )
+
+        return attrs
+
+    class Meta:
```
Comment (lilarin) [inline book/serializers.py:29]: Please, separate list view from the detail view, use child serializer for detail view, add here id field and remove it in child detail serializer

**Review by lilarin — CHANGES_REQUESTED** [review]

File: book/serializers.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            "author",
+            "cover",
+            "inventory",
+            "daily_fee"
+        )
+
+
+class BookDetailSerializer(BookSerializer):
+    class Meta(BookSerializer.Meta):
+        fields = ("title", "author", "inventory", "daily_fee")
```
Comment (lilarin) [inline book/serializers.py]: It is not necessary to use BookSerializer.Meta as parent class for Meta
Also, can you format fileds in similar style, as privious, please?

Comment (BornToLivee) [inline book/serializers.py]: cover?

**Review by lilarin — CHANGES_REQUESTED** [review]

**Review by lilarin — APPROVED** [review]
Nice

**Pull request merged**